Manuscript edits (#8)
petrelharp/num_edges: paper/writeup.tex

    it is an old observation that the ``junction'' between two distinct haplotypes
    produced by recombination is inherited as a Mendelian marker.
    In this paper, we describe how this recombination-mediated information
    can in many cases be recovered from inference based solely on polymorphic markers.
    % NSP-22NOV: "unary node" and "ARG" are not defined so I tried to provide better context here
    In a genealogical context, this information reflects the persistence
    of ancestral haplotypes across marginal genealogical trees where they
    are not coalescence events.

>     We show how these non-coalescing haplotypes (``unary nodes'') may be

Reviewer: I think we should call them something at least a bit different than "unary nodes" at least in the abstract, since "unary node" sounds like a node that is *always* unary.

Reviewer 2: "unary regions" should work I think

    inserted into the ancestral recombination graph (ARG), a compact
    but information-rich data structure describing the genealogical
    relationships among recombinant sequences.
    The resulting inferred ARGs are smaller, faster to compute with,

>     and contain substantially more information about inheritance % NSP-22NOV: "substantially more information" in what regard?

Reviewer: this is vague, which is I think okay, but "substantially more represented ancestral haplotype" in the sense of "proprtion truth represented"

recombination between distinct haplotypes can create a novel haplotype
whose relationships and origination time could be inferred.

Haplotype identity has been largely overlooked in the literature on ARG inference so far --
in that the metrics that have been used to measure accuracy of inferred ARGs
depend only on the sequence of marginal trees,
not on the sharing of ancestral haplotypes that span across these trees. % NSP-22NOV: what does "sharing of ancestral haplotype identities" mean, concretely? Would "... depend only on the sequence of marginal trees, but not on the ancestral haplotypes that span across these trees" be a clearer description?
% HF-24NOV: Noted and changed
Suppose two methods are used to infer ARGs from the same set of genotyped samples.
Although both ARGs describe the relationships between many haplotypes,
only those haplotypes that represent the focal set of genotyped samples % NSP-22NOV: the meaning here isn't clear to me-- what does "represent the focal set of genotyped samples" mean, concretely?

> % HF-24NOV : I am not exactly sure, I will ping Peter

Reviewer: I mean that if f() is a node-to-node mappings between ARGs, and `a` is a sample in the first ARG and `b` is the *same* sample in the second ARG, we want `f(a) = b`.

we define and describe how to compute new measures of (dis-)agreement between ARGs
that are motivated by the Robinson-Foulds distance between trees
but account for haplotype identity.

> These measures show that the vast majority of these extended haplotypes are correct, % NSP-22NOV: "vast majority" is correct in simulated (but not inferred) ARGs, right?
> % HF-24NOV : It should be both, as the IE line in Figure 6D shows that inferred trees match the TPR of the ground truth by at least 60% for inferred trees.

Reviewer: But the IE line in the other figure shows that a lot of extended stuff does not match also. So yes - this is for simulated trees; 
```suggestion
These measures show that the vast majority of these extended haplotypes are correct if the trees are correct,
```

Concretely, our method can be applied to any two tree sequences,
 as in the case of \figref{fig:conceptual_discrepancy},
 where we consider the pair of trees which exist at that break point.
However, to simplify notation suppose that the two tree sequences have the same set of breakpoints
between trees,
so that $T_1^{(1)}, \ldots, T_N^{(1)}$ are the trees in $\T_1$
and $T_1^{(2)}, \ldots, T_N^{(2)}$ are the trees in $\T_2$.
For a node $n$ and tree $T$
let $S(T, n)$ denote the set of samples that inherit from $n$ in $T$,
and for a pair of nodes $n_1$ and $n_2$ with $n_1$ in $\T_1$ and $n_2$ in $\T_2$,
define
% NSP-22NOV: The notation here is hard to read when subscripted. IMO it'd be better to use bracket notation here for the indicator function,
% $\mathbb{I}[ S(T_k^{(1)}, n_1) = S(T_k^{(2)}, n_2) ]$
% and also in next equation for consistency.

> % HF24-NOV: I think I disagree but we can get a third opinion.

Reviewer: Either way is okay by me, but a third option is to define in the text like 
\mathcal{M}(T1, n1; T2, n2) = \{ k : S(T^{(1)}_k,\ n_1) = S(T^{(2)}_k,\ n_2) \}
and then have the sum over this set.

\begin{align*}
    m(n_1, n_2)
    =
    \sum_{k=1}^N (a_k - a_{k-1}) \ind_{S(T^{(1)}_k,\ n_1) = S(T^{(2)}_k,\ n_2)}  

>     % NSP-22NOV: Added a missing comma. Also the comma at the very end looks like a prime, which is confusing, so I'm removing grammatical punctuation from these non-inline equations.

Reviewer: a better way would be to put some space (eg a `\quad`) in front of the comma

    % HF-24NOV: Noted
\end{align*}
which is the total span over which the samples below $n_1$ in $\T_1$
matches the samples below $n_2$ in $\T_2$.
Also let
\begin{align*}
    s(\T, n) = \sum_{k=1}^N (a_k - a_{k-1}) \ind_{n \in T_k} 
\end{align*}
the total span that node $n$ is present in the marginal trees.
The \emph{similarity} between $\T_1$ and $\T_2$ is then defined to be
\begin{align*}
    \similarity(\T_1, \T_2)
    =
    \max_{\beta:N_1 \to N_2} \sum_{n \in N_1} m(n, \beta(n)) 
\end{align*}

> where the maximum is over all mappings $\beta$ of nodes in $\T_1$ to nodes in $\T_2$.

Reviewer: whoops - should say "that maps samples to samples" (but, says this properly)

    \max_{\beta:N_1 \to N_2} \sum_{n \in N_1} m(n, \beta(n)) ,
    \max_{\beta:N_1 \to N_2} \sum_{n \in N_1} m(n, \beta(n)) 
\end{align*}
where the maximum is over all mappings of nodes in $\T_1$ to nodes in $\T_2$.
where the maximum is over all mappings $\beta$ of nodes in $\T_1$ to nodes in $\T_2$.
(Note that multiple nodes in $\T_1$ may be mapped to the same node in $\T_2$,
and that there may be no nodes in $\T_1$ that are mapped to some nodes in $\T_2$.)
Since the maximum is independent over nodes, we may define for each node $n_1 \in \T_1$
it's \emph{best matching node} in $\T_2$ as
\begin{align*}

>     \alpha(n_1) = \argmax_{n_2 \in N_2} m(n_1, n_2) ,

Reviewer: we don't want to remove punctuation here - punctuation is important!

to the minimum number of edge contraction/expansion operations needed to move
from one tree to the other
(which they then show is equal to the number of edges that induce different splits on the labels).
A similar metric on tree sequences could possibly be defined

> using subgraph-prune-and-regraft moves \citet{deng2024robust}.

Reviewer: ```suggestion
using the subgraph-prune-and-regraft moves used by \citet{deng2024robust}.
```
making it clear what the citation means

So far, we have demonstrated that there is potentially ample information
in the coalescent-only trees to extend haplotypes.
Does this work with \emph{inferred} ARGs?

> Indeed, most ARG inference methods provide inferred ARGS already containing unary nodes; % NSP-22NOV: this isn't true I don't think-- isn't it just tsinfer?
> % HF-24NOV: I am unsure. 

Reviewer: ```suggestion
Indeed, \tsinfer{} provides inferred ARGS already containing unary nodes;
```

(for instance, the MRCA of some set of sampled genomes).
Furthermore, \figref[A\&B]{fig:conceptual} implicitly takes the position
that relationships \emph{not} depicted in the ARG are implied to not exist.
An alternative interpretation
of the ARG depicted in \figref[A\&B]{fig:conceptual} would be that we have no information as to 
how node 2 inherited from node 4 on the right-hand interval,
rather than saying that the line of transmission specifically did not pass
through node 3.
The ``simplification'' algorithm of \citet{kelleher2018efficient},
or the Hudson algorithm for coalescent simulation
as implemented in msprime \citep{kelleher2016efficient},

> each specifically discard information about any such ``non-coalescent'' nodes;

Reviewer: ```suggestion
each specifically discard information about any such ``non-coalescent'' portions of ancestral haplotypes;
```

of the true ancestry of a set of samples (i.e., all their genetic ancestors who ever lived),
and so we wanted to quantify
``How much of the true relationships does this ARG represent?''
However, it is worth noting that the symmetrized version
$d(\T_1,\T_2) = \dis(\T_1,\T_2) + \dis(\T_2, \T_1)$ is a metric.
To see this, first suppose that we have a bijection between the nodes of $\T_1$ and $\T_2$,
and view each ARG as a subset of the space $[0,L) \times N \times N$,
where $N$ is the shared set of nodes.
Then, dissimilarity is the Lebesgue measure of the relative difference of the two sets:
$|\T_1 \setminus \T_2|$,
and so the symmetrized version is the measure of the symmetric difference
$|\T_1\Delta\T_2|$,
which is well-known to be a metric. % NSP-22NOV: "well-known" ==> give a citation here

> % HF-24NOV: Peter would we cite Rudin?

Reviewer: I don't know - have you looked in Rubin? (But - I agree a citation would be nice if we have one handy, but it's not totally necessary.) It's [in wikipedia](https://en.wikipedia.org/wiki/Symmetric_difference#Symmetric_difference_on_measure_spaces) after all?

If the two ARGs have the same number of nodes,
we can consider all bijections between their nodes.
The symmetric difference between $\T_1$ and $\T_2$,
related through each bijection is a metric.
Then the minimum over all such metrics will still be a metric
since the minimum over a finite number of metrics is also a metric.
 % NSP-22NOV: I don't understand this sentence, can you rewrite to be clearer?
 % HF-24NOV: Hopefully it is clearer now

> This also extends to two tree sequences with different numbers of nodes, $|N_1|\neq|N_2|$, as we can take the minimum over all possible matchings $T_1\to\T_2$ and $\T_2\to\T_1$.

Reviewer: I'm not quite following this - given two functions `f : [N_1] \to [N_2]` and `g : [N_2] \to [N_1]`, I think you're saying the size of the symmetric difference is a metric, but on what? The size of the symmetric difference is a mapping from (ARGs with N_1 nodes) \times (ARGs with N_2 nodes) to the nonnegative reals, but a metric needs to be on X \times X for some space X. Hm, I suppose you can just define all ARGs to be on countably many nodes, but most of which don't have any relationships between them. Ah, yes, this works, I think! And, a nice way to deal with this.

Reviewer 2: Yeah if you consider the ambient space N_1 \times N_2 where both T_1 and T_2 are canonically embedded it just holds for the exact same reason the symmetric difference itself is a metric.

Clearly, there is ample information in genomic data to infer the extent of ancestral haplotypes even where they are unary,
and this has the potential to significantly inform downstream methods, especially dating.
However, currently not all ARG inference methods return ARGs with unary nodes.
For instance, we have seen that tsinfer returns ARGs to which the method presented here adds little -
For instance, we have seen that tsinfer returns ARGs to which the method presented here adds little --
so, seems to be already taking advantage of this source of information.
Singer [cite] returns ARGs without any unary nodes,
although these are reconstructable using the times of the recombination nodes.

> On the other hand, locally building trees separately in each segment of genome

Reviewer: This was based on a conversation wtih Yun Deng on slack - so, it'd be nice to insert it, unless we think it's wrong? (And, it would be good to cite ARGNeedle somewhere, ideally earlier...)

Author: As far as I know, SINGER doesn't return recombination nodes either, so this seemed like a confusing and unnecessary level of detail. (But, maybe this has changed in a recent version of SINGER?) If you think it's important, I'm fine w/ adding it back it

Reviewer 3: I'd be really surprised if SINGER doesn't return recombination nodes - that's the model it's inferring after all.

Author: It doesn't stick them into the output tree sequence, only coalescing nodes (at least last time I looked, a couple months ago). They may be stored in its own internal format, but there's no documentation on this as far as I know.
